src/main/java/core/basesyntax/operations/PurchaseOperation.java

```
@@ -0,0 +1,26 @@
+package core.basesyntax.operations;
+
+import core.basesyntax.FruitTransaction;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public Map<String, Integer> getCalculation(
+            Map<String, Integer> fruits, FruitTransaction fruitTransaction) {
+        if (fruits.isEmpty()) {
+            throw new RuntimeException("Fruit's list is empty");
+        }
+        return fruits.entrySet().stream()
+                .map(entry -> {
+                    int newValue = entry.getKey().equals(fruitTransaction.getFruit())
+                            ? entry.getValue() - fruitTransaction.getQuantity()
+                            : entry.getValue();
+                    if (newValue < 0) {
+                        throw new RuntimeException("Value cant be smaller than 0");
+                    }
+                    return Map.entry(entry.getKey(), newValue);
+                })
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));

```

Issue: This block creates and returns a new map using streams, rather than updating the provided `fruits` map in place. According to the checklist, operation handlers should modify the existing map directly. Please update the code to perform the subtraction directly on the `fruits` map.

Additionally, there is no check to ensure the fruit exists in the map before attempting to subtract the quantity. You should add a check to throw an exception if the fruit is missing, as required by the checklist.